karabo_data/reader.py

```diff
@@ -994,6 +995,47 @@ def write_virtual(self, filename):
         from .writer import VirtualFileWriter
         VirtualFileWriter(filename, self).write()
 
+    def get_virtual_dataset(self, source, key):
+        """Create an HDF5 virtual dataset for a given source & key
+
+        A dataset looks like a multidimensional array, but the data is loaded
+        on-demand when you access it. So it's suitable as an
+        interface to data which is too big to load entirely into memory.
+
+        This returns an h5py.Dataset object. This exists in a real temporary
+        file as a 'virtual dataset', a collection of links pointing to the
+        data in real datasets. The file path is available as ``ds.file.name``,
+        and the path of the dataset inside the file as ``ds.name``.
+
+        The temporary file may be cleaned up when this process exits, and will
+        probably not be visible from other machines. If you want to access the
+        data for longer or from other systems, for instance for distributed
+        data processing, you can copy the temporary file to another location.
+        It should be a small file, as it only points to the real data. However,
+        it will only be usable where you have access to the original data files.
+        """
+        self._check_field(source, key)
+
+        from .writer import VirtualFileWriter
+        fd, filename = tempfile.mkstemp(suffix='-karabo-data-vds.h5')
+        os.close(fd)
+        vfw = VirtualFileWriter(filename, self)
```

Comment:
Is there something preventing us to use the `core` driver + no backing store? That could be used to avoid creating a file at all (file would exists only in memory)

Reply:
I'm considering it a feature that there's a real file. My thinking is that you can pass the path and dataset name to worker processes, which would open the file to access the data.

I'm considering having an optional `filename=` parameter to specify where the file should be created, and using the tempfile if it's not given.

Reply:
+1 for the in memory option using the core driver. I had some cases where users wanted to use virtual datasets to do easier processing of detector data across trains. So I guess creating a memory view of a virtual dataset could be of value. Think about it.

Reply:
There's not much user-visible difference between a purely in-memory object and creating a small temp file to hold the virtual dataset. Since we want a real file for some use cases anyway, I'm inclined to stick with the default file driver, and avoid any obscure bugs that might come from using the core driver.